cadcdata/cadcdata/storageinv.py

```diff
@@ -0,0 +1,826 @@
+        if self._cadc_client.subject.anon:
+            raise exceptions.UnauthorizedException(
+                'Must be authenticated to remove data')
+
+        urls = self._get_transfer_urls(id, is_get=False)
+        if len(urls) == 0:
+            raise exceptions.NotFoundException(
+                'File not found: {}'.format(id))
+
+        # get the list of transfer points
+        for url in urls:
+            logger.debug(
+                'REMOVE file with identifier {} from URL {}'.format(id, url))
+            try:
+                start = time.time()
+                self._cadc_client.delete(url)
+                duration = time.time() - start
+                logger.debug('{} removed in {} ms'.format(id, duration))
+                return
+            except (exceptions.HttpException, socket.timeout) as e:
+                # try a different URL
+                logger.info('WARN: Cannot remove data from {}. Exception: {}'.
+                            format(url, e))
+                logger.warn('Try the next URL')
+                continue
+        raise exceptions.HttpException(
+            'Unable to remove data from any of the available URLs')
+
+    def cadcinfo(self, id):
+        """
+        Get information regarding a file in the archive
+        :param archive: Name of the archive
+        :param file_name: name of the file
+        :returns dictionary of attributes/values
+        """
+        resource = (FILES_STANDARD_ID, id)
+        logger.debug('HEAD {}'.format(resource))
+        try:
+            response = self._cadc_client.head(resource, allow_redirects=True)
+        except exceptions.NotFoundException as e:
+            raise exceptions.NotFoundException(id, e)
+        h = response.headers
+        file_info = FileInfo(id)
+        size = h.get('Content-Length', None)
+        if size is not None:
+            file_info.size = int(size)
+        file_info.md5sum = net.extract_md5(response.headers)
+        file_info.name = net.netutils.get_header_filename(response.headers)
+        if h.get('Last-Modified', None):
+            file_info.lastmod = \
+                datetime.datetime.strptime(h.get('Last-Modified'),
+                                           '%a, %d %b %Y %I:%M:%S %Z')
+        file_info.file_type = h.get('Content-Type', None)
+        file_info.encoding = h.get('Content-Encoding', None)
+        logger.debug('File info: {}'.format(file_info))
+        return file_info
+
+    def _get_transfer_urls(self, id, is_get=True, headers=None, params=None):
+        if not self.transfer:
+            # this is site location
+            return ['{}/{}'.format(self.files, id)]
+        trans = net.Transfer(self._cadc_client._get_session())
+        return trans.transfer(
+            endpoint_url=self.transfer, uri=id,
+            direction='pullFromVoSpace' if is_get else 'pushToVoSpace',
+            with_uws_job=False)
+
+    def _get_md5sum(self, filename):
```

Comment:
Is this method still required?

Reply:
For large files this is costly but for smaller ones it could be advantageous to precompute the md5sum and send it in the initial request. I don't know where the threshold is but I'd like to keep it around until I do the next story about `PUT` with transactions.